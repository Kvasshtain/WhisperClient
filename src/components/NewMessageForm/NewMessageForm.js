import React from 'react'
import PropTypes from 'prop-types'

import './NewMessageForm.sass'

class NewMessageForm extends React.Component {
  state = {
    newMessage: '',
  }

  onSubmit = event => {
    event.preventDefault()

    this.props.onSubmitNewMessage(this.state.newMessage)

    this.setState({
      newMessage: '',
    })
  }

  updateInputValue = eventArg => {
    this.setState({
      newMessage: eventArg.target.value,
    })
  }

  render() {
    return (
      <form className="newMessageForm" onSubmit={this.onSubmit}>
        <h4>+Message</h4>
        <input
          name="userMessage"
          placeholder="message text"
          type="text"
          value={this.state.newMessage}
          onChange={this.updateInputValue}
        />
        <button type="submit">Send message</button>
      </form>
    )
  }
}

NewMessageForm.propTypes = {
  onSubmitNewMessage: PropTypes.func.isRequired,
}

export { NewMessageForm }
