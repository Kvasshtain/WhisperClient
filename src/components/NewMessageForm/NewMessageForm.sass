.newMessageForm
    padding: 10px
    background: linear-gradient(to bottom, #f5feea, #ace6ba)
    border-top: 1px solid #afafaf