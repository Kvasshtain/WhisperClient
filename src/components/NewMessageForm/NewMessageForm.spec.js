import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { NewMessageForm } from './NewMessageForm'

describe('NewMessageForm', () => {
  const mockOnSubmitNewMessage = jest.fn()

  it('should render correctly', () => {
    const testedComponent = shallow(
      <NewMessageForm onSubmitNewMessage={mockOnSubmitNewMessage} />
    )

    expect(shallowToJson(testedComponent)).toMatchSnapshot()
  })
})

describe('when typing new message', () => {
  const mockOnSubmitNewMessage = jest.fn()
  const newTestMessage = 'Some new message'

  const testedComponent = shallow(
    <NewMessageForm onSubmitNewMessage={mockOnSubmitNewMessage} />
  )

  beforeEach(() => {
    testedComponent.find('input').simulate('change', {
      target: {
        value: newTestMessage,
      },
    })
  })

  it('updates inputValue field in state', () => {
    expect(testedComponent.state().newMessage).toEqual(newTestMessage)
  })
})

describe('when push the button', () => {
  const mockOnSubmitNewMessage = jest.fn()

  const myComponent = shallow(
    <NewMessageForm onSubmitNewMessage={mockOnSubmitNewMessage} />
  )

  beforeEach(() => {
    myComponent.find('form').simulate('submit', {
      preventDefault: () => {},
    })
  })

  it('calls the props.onSubmit', () => {
    expect(mockOnSubmitNewMessage).toHaveBeenCalledTimes(1)
  })
})
