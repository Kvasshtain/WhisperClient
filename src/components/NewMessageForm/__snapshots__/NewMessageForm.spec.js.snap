// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessageForm should render correctly 1`] = `
<form
  className="newMessageForm"
  onSubmit={[Function]}
>
  <h4>
    +Message
  </h4>
  <input
    name="userMessage"
    onChange={[Function]}
    placeholder="message text"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Send message
  </button>
</form>
`;
